apollo-router/src/cache/storage.rs

```diff
@@ -8,7 +8,7 @@ use tokio::sync::Mutex;
 //
 // this will be replaced by the multi level (in memory + redis/memcached) once we find
 // a suitable implementation.
-#[derive(Clone)]
+#[derive(Clone, Debug)]
```

Comment:
NIT: or...
```
diff --git a/apollo-router/src/cache/storage.rs b/apollo-router/src/cache/storage.rs
index 447202ff..2eb6a08c 100644
--- a/apollo-router/src/cache/storage.rs
+++ b/apollo-router/src/cache/storage.rs
@@ -8,7 +8,7 @@ use tokio::sync::Mutex;
 //
 // this will be replaced by the multi level (in memory + redis/memcached) once we find
 // a suitable implementation.
-#[derive(Clone, Debug)]
+#[derive(Clone)]
 pub(crate) struct CacheStorage<K: Hash + Eq + Send, V: Clone> {
     inner: Arc<Mutex<LruCache<K, V>>>,
 }
diff --git a/apollo-router/src/introspection.rs b/apollo-router/src/introspection.rs
index e069e2a7..515a0307 100644
--- a/apollo-router/src/introspection.rs
+++ b/apollo-router/src/introspection.rs
@@ -10,7 +10,6 @@ use crate::graphql::Response;
 const DEFAULT_INTROSPECTION_CACHE_CAPACITY: usize = 5;
 
 /// A cache containing our well known introspection queries.
-#[derive(Debug)]
 pub(crate) struct Introspection {
     cache: CacheStorage<String, Response>,
 }
diff --git a/apollo-router/src/query_planner/bridge_query_planner.rs b/apollo-router/src/query_planner/bridge_query_planner.rs
index 6e01e2f1..af997a1f 100644
--- a/apollo-router/src/query_planner/bridge_query_planner.rs
+++ b/apollo-router/src/query_planner/bridge_query_planner.rs
@@ -24,7 +24,7 @@ use crate::*;
 
 pub(crate) static USAGE_REPORTING: &str = "apollo_telemetry::usage_reporting";
 
-#[derive(Debug, Clone)]
+#[derive(Clone)]
 /// A query planner that calls out to the nodejs router-bridge query planner.
 ///
 /// No caching is performed. To cache, wrap in a [`CachingQueryPlanner`].
```

I'm not against propagating the `Debug` around, but if we aren't using it anywhere (and it doesn't seem like we are), then maybe we should consider removing that constraint?

Reply:
Good call, I wasn't sure it wasn't used anywhere, but if I remove it from here, i ll also remove it from `BridgeQueryPlanner` but tbh i don't see any reason why it would need to derive debug anyway

Reply:
fixed in [34024e3](https://github.com/apollographql/router/pull/1517/commits/34024e31634738b0be6a8679c0b49b95c98813c6)